### ISSUE TITLE
Infinite Loop when not authorized or forbidden

### ISSUE THREAD

**Issue by patricksimonian** [GatsbyCentral/gatsby-source-eventbrite#18]
If i use an access token that results in 401 or 403 errors, the plugin will loop infinitely.

Looks like the `fetch.js` loop has no provision for breaking out of the loop during a catch

**Comment by kwelch** [comment]
Hey thanks for pointing this out. 

Where did you see the loop?

I was able to get it to error with a `401` by providing a bad token, but I was not seeing it loop at all. It just threw an error in the build step, it also errors in dev, but I was not able to reproduce the loop effect. 

Whoops! My bad, this is not the repo I thought it was. 🤦‍♂️ I also have an Eventbrite source plugin.

**Comment by patricksimonian** [comment]
oh hahah! Yes i took a look at yours! I definitely agree that the two projects should be merged at some point. I"ve been playing with event brites api myself and have found it a bit clunky. Especially with scoped access tokens. It appears they don't have that? For example, i'm apart of an eventbrite team but only the owner of the account's access token works to pull org data.

**Comment by kwelch** [comment]
Interesting, I know about the org work but I haven't used it much. I will take this back to the team internally and see if there are any efforts to improve that experience. 

Let me know if there is any way I can help.

**Comment by patricksimonian** [comment]
Absolutely, is there a preferred channel to communicate with you and the team instead of this issue?

**Comment by kwelch** [comment]
You can DM me on twitter `@kylewelch`.
